Frontend/src/Components/Battery/Battery.js

```diff
@@ -0,0 +1,140 @@
+import {Flex, Center, Box, Spacer, Text, useColorMode} from "@chakra-ui/react";
+import RangeRow from "../Shared/RangeRow";
+import RangeBar from "../Shared/RangeBar";
+import CONSTANTS from "../../data-constants.json";
+import getColor from "../Shared/colors";
+
+export default function Battery(props) {
+    const { colorMode } = useColorMode();
+    const headerBg = getColor("header", colorMode);
+    const borderColor = getColor("border", colorMode);
+
+    const fanEnableText = (props.data?.fan_enable[0] ?? -1.0) ? 'ON' : 'OFF';
+    const fanEnableColor = (props.data?.fan_enable[0] ?? -1.0) ? '#89FC4D' : '#F5C2C1';
+
+    const chargeEnableText = (props.data?.charge_enable[0] ?? -1.0) ? 'Enable' : 'Disable';
+    const chargeEnableColor = (props.data?.charge_enable[0] ?? -1.0) ? '#89FC4D' : '#F5C2C1';
+
+    return (
+        <Flex flex="auto" direction="column" overflowY="scroll">
+            <RangeRow
+                dataTitle="Pack Temp"
+                dataValue={props.data?.pack_temp[0] ?? -1.0}
+                dataConstant={CONSTANTS.pack_temp}
+            />
+            <RangeRow
+                dataTitle="Pack Internal Temp"
+                dataValue={props.data?.pack_internal_temp[0] ?? -1.0}
+                dataConstant={CONSTANTS.pack_internal_temp}
+
+                bg={headerBg}
+            />
+            <Center
+                borderTopWidth={1}
+                borderBottomWidth={1}
+                borderRightWidth={1}
+                borderColor={borderColor}
+                lineHeight='1.2em'
+                height='100%'
+            >
+                <Flex w='95%'>
+                    <Flex direction='column'>
+                        <Spacer />
+                        <Box
+                            border={"1px"}
+                            borderRadius='0px'
+                            h='1em'
+                            background={fanEnableColor}
+                        >
+                            <Text fontSize='xs' margin='0.05px'>{fanEnableText}</Text>
+                        </Box>
+                        <Spacer />
+                    </Flex>
+                    <Spacer />
+                    <Text fontSize='sm'>Fan Speed:</Text>
+                    <Spacer />
+                    <Text fontSize='sm' paddingRight='0.5em'>
+                        {(props.data?.fan_speed[0] ?? -1.0).toFixed(0)} {CONSTANTS.fan_speed.UNIT}
+                    </Text>
+
+                    {/* Hold the RangeBar within a vertical Flex to vertically center it */}
+                    <Flex direction='column'>
+                        <Spacer />
+                        <RangeBar 
+                            w={props.rangeBarW ?? '3em'} 
+                            h='1em'
+                            borderRadius='0px' 
+
+                            val={(props.data?.fan_speed[0] ?? -1.0)}
+                            min={0}
+                            max={100}
```

Comment:
Similar to the comment above, you can reference `BatteryCells/BatteryPack.js` for determining the `fan_speed` nominal minimum and maximum values (probably worth finding/setting the minimum value outside of the return statement so that the logic doesn't have to be included twice).

Reply:
Would it be good to have a comment explaining how we got 

```jsx
((props.data?.pack_temp[0] ?? -1.0) < (20 + 5 * CONSTANTS.fan_speed.MAX)) ?
                         Math.floor(((((props.data?.pack_temp[0] ?? -1.0) >= 20) ? props.data?.pack_temp[0] : 15) - 15) / 5) :
                         CONSTANTS.fan_speed.MAX
```

as the way that fan speed min is calculated?

Reply:
Yes, that would be a good idea. For the comment, you can use what we have in the signal spreadsheet, plus a statement about the maximum value of fanSpeedMin: "For every 5˚C increment in pack temp above 20˚C, fan speed should increase by 1. However, fanSpeedMin cannot be greater than the nominal maximum value of fan_speed"